src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -50,6 +53,62 @@ internal class PackageLoadContext
 
         public Task<PackageCollection> GetInstalledPackagesAsync() =>_installedPackagesTask;
 
+        // get lock file for projects
+        public async Task<List<LockFile>> getLockFiles(IEnumerable<NuGetProject> nuGetProjects)
+        {
+            var nuGetProjectList = nuGetProjects.ToList();
+            var buildIntegratedProjects = nuGetProjectList.OfType<BuildIntegratedNuGetProject>().ToList();
+
+            List<LockFile> lockFiles = new List<LockFile>();
+            var lockFileFormat = new LockFileFormat();
+
+            foreach (var project in buildIntegratedProjects)
+            {
+                var lockFilePath = await project.GetAssetsFilePathAsync();
+                if (File.Exists(lockFilePath))
+                {
+                    lockFiles = lockFiles.Append(lockFileFormat.Read(lockFilePath)).ToList();
+                }
+            }
+            return lockFiles;
+        }
+
+        // get the dependent packages from the lock files for projects
+        public async Task<PackageCollection> GetDependentPackagesAsync()
+        {
+            // get lock files from projects. This will add all dependency packages for all projects in the list
+            List<LockFile> lockFiles = await getLockFiles(Projects.ToList());
+            List<PackageCollectionItem> dependentPackages = new List<PackageCollectionItem>();
+
+            foreach (LockFile lockFile in lockFiles)
+            {
+                foreach (LockFileTarget target in lockFile.Targets)
+                {
+                    foreach(LockFileTargetLibrary lib in target.Libraries)
+                    {
+                        foreach (Packaging.Core.PackageDependency dep in lib.Dependencies)
+                        {
+                            Versioning.NuGetVersion version;
```

Comment:
Why do we need only one version?
We shouldn't use Versioning.NuGetVersion, we should do the correct using just refer to it as NuGetVersion.
We only use namespaces when there are naming clashes.

Reply:
Now returning version range. 
Added using for Versioning.